### PR TITLE
[DEV-3234] Update created table ownership

### PR DESCRIPTION
**Pull request by kchua78**
## Description

Update the owner of created tables for Databricks Unity

## Type of Change

Label this pull request to place it under the correct category in Release Notes:

|        **Pull Request Label**         | **Category in Release Notes** |
|:-------------------------------------:|:-----------------------------:|
|       `enhancement`, `feature`        |          🚀 Features          |
| `bug`, `refactoring`, `bugfix`, `fix` |    🔧 Fixes & Refactoring     |
|       `build`, `ci`, `testing`        |    📦 Build System & CI/CD    |
|              `breaking`               |      💥 Breaking Changes      |
|            `documentation`            |       📝 Documentation        |
|            `dependencies`             |    ⬆️ Dependencies updates    |

## Checklist

- [ ] I have read the [`CODE_OF_CONDUCT.md`](https://github.com/featurebyte/featurebyte/blob/main/CODE_OF_CONDUCT.md) and [`CONTRIBUTING.md`](https://github.com/featurebyte/featurebyte/blob/main/CONTRIBUTING.md) guides.
- [ ] I have written tests for the changes made.
- [ ] I have written docstrings in [NumpyDoc format](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)
- [ ] I have labeled my Pull Request correctly

### CODE REVIEW

File: featurebyte/session/databricks_unity.py [hunk 1]
```diff
@@ -129,3 +133,21 @@ async def list_tables(
...
+        kind: Literal["TABLE", "VIEW"] = "TABLE",
+        replace: bool = False,
+    ) -> pd.DataFrame | None:
+        result = await super().create_table_as(table_details, select_expr, kind, replace)
+        # grant ownership of the table to the group
+        fully_qualified_table_name = sql_to_string(
+            get_fully_qualified_table_name(table_details.dict()), source_type=self.source_type
+        )
+        grant_permissions_query = (
+            f"ALTER TABLE {fully_qualified_table_name} OWNER TO `{self.group_name}`"
```
Comment (YS-L) [inline featurebyte/session/databricks_unity.py:150]: Should `fully_qualified_table_name` be quoted for consistency?

Comment (YS-L) [reply featurebyte/session/databricks_unity.py:150]: Oh I think it's quoted already.

**Review by YS-L — APPROVED** [review]

File: poetry.lock [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# This file is automatically @generated by Poetry 1.8.2 and should not be changed by hand.
+# This file is automatically @generated by Poetry 1.5.1 and should not be changed by hand.
```
Comment (gee-senbong) [inline poetry.lock:1]: Are these changes expected? Should we revert changes here?

Comment (kchua78) [reply poetry.lock:1]: this is to update idna to fix reported vulnerabilities

File: featurebyte/session/databricks_unity.py [hunk 3]
```diff
@@ -129,3 +176,39 @@ async def list_tables(
...
+            table_details = TableDetails(
+                database_name=None, schema_name=None, table_name=table_details
+            )
+        fully_qualified_table_name = sql_to_string(
+            get_fully_qualified_table_name(table_details.dict()), source_type=self.source_type
+        )
+        grant_permissions_query = (
+            f"ALTER {kind} {fully_qualified_table_name} OWNER TO `{self.group_name}`"
+        )
+        await self.execute_query(grant_permissions_query)
```
Comment (YS-L) [inline featurebyte/session/databricks_unity.py:213]: Would it be useful to have a helper method to alter owner? Would be easier to make changes later if it is in a centralised place.

**Review by YS-L — APPROVED** [review]

**Pull request merged**